src/MR.EntityFrameworkCore.KeysetPagination/KeysetFilterPredicateStrategy.cs

```diff
@@ -92,11 +92,23 @@ internal abstract class KeysetFilterPredicateStrategy : IKeysetFilterPredicateSt
 		else
 		{
 			return compare(
-				memberAccess,
-				EnsureMatchingType(memberAccess, referenceValue));
+				EnsureEnumIsConvertedToUnderlyingType(memberAccess),
+				EnsureEnumIsConvertedToUnderlyingType(EnsureMatchingType(memberAccess, referenceValue)));
 		}
 	}
 
+	private static Expression EnsureEnumIsConvertedToUnderlyingType(Expression memberAccess)
```

Comment:
Would be better to call this something more generic rather than tie it to enum only conversions. Also better to rename `memberAccess` to just `expression` as it might not be a member access in the first place (and we don't need to know here).

```suggestion
	private static Expression EnsureAdditionalConversions(Expression expression)
```